File: interfaces/java/com/kx/Month.java
package com.kx;

import static com.kx.Type.NullInteger;
import static com.kx.Type.toTwoDigitString;

/**
 * This class is a kdb+ month.
 * 
 * @see <a href="https://code.kx.com/trac/wiki/Cookbook/InterfacingWithJava">Interfacing With Java at code.kx</a> 
 * @author tech@kx.com
 */
public class Month {
	
	private int month;

	/**
	 * Constructs a kdb+ month.
	 * 
	 * @param month
	 */
	public Month(int month) {
		this.month = month;
	}

	/**
	 * Converts to a string.
	 *
	 * @see java.lang.Object#toString()
	 */
	@Override	
	public String toString() {
		int m = month + 24000, y = m / 12;
		if (month == NullInteger) 
			return "";
		else 
			return toTwoDigitString(y / 100) + toTwoDigitString(y % 100) 
				+ "-"
				+ toTwoDigitString(1 + m % 12);
	}

	/**
	 * Checks for equality.
	 *
	 * @see java.lang.Object#equals(Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (obj instanceof Month) 
			return (month == ((Month)obj).month);
		else 
			return false;
	}

	/**
	 * Gets the hashcode.
	 *
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		return month;
	}

	/**
	 * Gets the month.
	 * 
	 * @return the month
	 */
	public int getMonth() {
		return month;
	}

	/**
	 * Sets the month.
	 * 
	 * @param month
	 */
	public void setMonth(int month) {
		this.month = month;
	}
	
}

File: interfaces/csharp/Kdbplus/Month.cs
namespace Kdbplus
{
    /// <summary>
    /// This is a kdb+ month value.
    /// </summary>
    public class Month
    {
        private int theMonth;

        /// <summary>
        /// Constructs based on the given int value.
        /// </summary>
        /// <param name="m"></param>
        public Month(int m)
        {
            theMonth = m;
        }

        /// <summary>
        /// Gets and sets the month value.
        /// </summary>
        public int TheMonth
        {
            get { return theMonth; }
            set { theMonth = value; }
        }

        /// <summary>
        /// Converts to a string.
        /// </summary>
        /// <returns>The string representation of this value.</returns>
        public override string ToString()
        {
            int m = 24000 + theMonth, y = m/12;
            return theMonth == Type.NullInteger ? "" : Type.ToTwoDigitString(y/100) + Type.ToTwoDigitString(y%100) + "-" + Type.ToTwoDigitString(1 + m%12);
        }
    }
}